### PR TITLE
filter user e-mail in graph/user requests

### PR DESCRIPTION
**Pull request by dragonchaser**
Bugfix: Mask user email in output

We have fixed a bug where the user email was not masked in the output and the user emails could be enumerated through
the sharee search. This is the ocis side which adds an suiting config option to mask user emails in the output.

refs #8726

### CODE REVIEW

**Review by kobergj — APPROVED** [review]

File: services/graph/pkg/config/reva.go [hunk 1]
```diff
@@ -2,5 +2,6 @@ package config
 
 // TokenManager is the config for using the reva token manager
 type TokenManager struct {
-	JWTSecret string `yaml:"jwt_secret" env:"OCIS_JWT_SECRET;GRAPH_JWT_SECRET" desc:"The secret to mint and validate jwt tokens." introductionVersion:"pre5.0"`
+	JWTSecret              string `yaml:"jwt_secret" env:"OCIS_JWT_SECRET;GRAPH_JWT_SECRET" desc:"The secret to mint and validate jwt tokens." introductionVersion:"pre5.0"`
+	ShowUserEmailInResults bool   `yaml:"mask_user_email" env:"OCIS_SHOW_USER_EMAIL_IN_RESULTS" desc:"Mask user email addresses in responses." introductionVersion:"5.1"`
```
Comment (rhafer) [inline services/graph/pkg/config/reva.go:6]: Why did you put the setting in the `TokenManager` struct. I don't think it should go there.

Wouldn't it better fit into `config.API`? See config.go

The same it true for the `ocs` service btw.

**Review by rhafer — CHANGES_REQUESTED** [review]
I think the config option is misplaced. We should also fix that for `ocs`. (Also I somewhat disagree on the requirement to have this configurable at all. We already have far too many knobs. But I guess that is not my call)

File: services/graph/pkg/config/reva.go (hunk 1 above)
Comment (kobergj) [reply services/graph/pkg/config/reva.go:6]: Oh, if it is still for debate if we need this switch or not, I am team Ralf :smiley:

Comment (dragonchaser) [reply services/graph/pkg/config/reva.go]: I already adapted the code

**Review by rhafer — APPROVED** [review]

**Pull request merged**